### PR TITLE
Update the spinner typings to accept a custom label component

### PR DESCRIPTION
**Pull request by sradevski**
Change-type: patch
Signed-off-by: Stevche Radevski <stevche@balena.io>

---
##### Contributor checklist

- [ ] I have rebuilt the README with `npm run build:docs`
- [ ] I have regenerated screenshots for any affected components with `npm run generate-screenshots`
- [ ] I have regenerated jest snapshots for any affected components with `npm run jest -- -u`

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -676,7 +676,7 @@ You can refer to the [grommet select page](https://v2.grommet.io/select) for all
 | `children`     | `React.ReactNode` |         | -          | If the spinner has children, it will show an opaque overlay over the children when it is spinning |
 | `show`         | `boolean`         | true    | -          | If passed, it will control whether the spinner is shown or not |
 | `emphasized`   | `boolean`         | false   | -          | If true, it will render a large spinner |
-| `label`        | `string`          |         | -          | A label that will be rendered next to the spinner. Renders on right-hand side for standard spinner, and below spinner for emphasized |
+| `label`        | `string` | `React.ReactNode`|            | -          | A label that will be rendered next to the spinner. Renders on right-hand side for standard spinner, and below spinner for emphasized |
```
Comment (iamsolankiamit) [inline README.md]: the `|` between `string` and `React.ReactNode` is messing with the table layout. we can either use `&#124;` or `\|`

Comment (sradevski) [reply README.md]: It was my bad, they all had to be as a code snippet

Comment (iamsolankiamit) [reply README.md]: wrapping it with `` ` `` (backtick) won't fix it

for ex. The table  has `a|b` and it doesn't works

| a |  ab     | c |
|---|---------|---|
| a |  `a|b`  | c |

and the table below has `a\|b` and it does works

| a |   ab   | c |
|---|--------|---|
| a |  a\|b  | c |

**Comment by sradevski** [comment]
@balena-ci retest

File: README.md (hunk 1 above)
Comment (sradevski) [reply README.md]: @iamsolankiamit In other places we also use the code snippet pattern and it renders as expected in VSCode for example. I agree that what you suggest is better though, I'll update now.

**Pull request merged**